src/utils/trpc.ts

```diff
@@ -79,14 +90,13 @@ export const trpc = createTRPCNext<AppRouter, SSRContext>({
         defaultOptions: {
           queries: {
             staleTime: 1000 * 10, // 10 seconds
+            useErrorBoundary: true,
             retry: (failureCount, error) => {
               if (error instanceof TRPCClientError) {
-                if (error.data?.code === 'FORBIDDEN') {
-                  Router.push('/forbidden')
-                }
+                const res = TRPCWithErrorCodeSchema.safeParse(error)
 
-                if (error.data?.code === 'UNAUTHORIZED') {
-                  Router.push('/sign-in')
+                if (res.success && NON_RETRYABLE_ERROR_CODES.has(res.data)) {
```

Comment:
seems like a regression here, should we redirect to the appropriate pages if needed?

Reply:
Using `Router.push` here will not work if an error was thrown and the top level error boundary catches the error.

Instead of relying on router.push, im thinking we can set a convention to do all redirects in the `ErrorBoundary` component instead.

It's a little abstract, but we can walk through it later and I can demo what i mean.

to reproduce the behaviour, remove the unauthorized relocation block from `ErrorBoundary`, and try to redirect to `sign-in` from the `trpc` singleton. You'll see that the redirection does not load the `Sign-In` page as the component has not been mounted.

This is an edge case that only happens if the top level `ErrorBoundary` catches something and does not render the page with layout component